hpcbench/driver.py

```diff
@@ -696,14 +677,48 @@ def __call__(self, **kwargs):
             extractors = all_extractors
         if not isinstance(extractors, list):
             extractors = [extractors]
-        metrics = self.report.setdefault('metrics', {})
-        for extractor in extractors:
-            run_metrics = extractor.extract(os.getcwd(),
-                                            self.report.get('metas'))
-            MetricsDriver._check_metrics(extractor.metrics, run_metrics)
-            metrics.update(run_metrics)
+        all_metrics = self.report.setdefault('metrics', [])
+        for log in self.logs:
+            metrics = {}
+            for extractor in extractors:
+                with extractor.context(log.path, log.log_prefix):
+                    run_metrics = extractor.extract(self.report.get('metas'))
+                    MetricsDriver._check_metrics(extractor.metrics,
+                                                 run_metrics)
+                    metrics.update(run_metrics)
+            rc = dict(context=log.context, measurement=metrics)
+            all_metrics.append(rc)
         return self.report
 
+    @property
+    def logs(self):
+
+        class LocalLog(namedtuple('LocalLog', ['path', 'log_prefix'])):
```

Comment:
maybe a stupid question, but `logs` will be called plenty of times, every time re-creating `LocalLog` and `SrunLog` classes, doesn't that produce quite a bit of overhead that could be avoided by moving these two classes somewhere outside

Reply:
Overhead is minor, but good point though. I moved those 2 classes definition inside `MetricsDriver` class definition.